packages/components/src/mobile-promo-card/index.tsx

```diff
@@ -13,42 +14,110 @@ export type MobilePromoCardProps = {
 	isWoo?: boolean;
 };
 
+// Slugs as used by Jetpack Redirects.
+// See https://jetpack.com/redirect for current URLs.
+// Two _QRCODE constants are included for reference only.
+
+const URL_JETPACK_A8C = 'calypso-stats-mobile-cta-jetpack-link';
+const URL_JETPACK_APPLE = 'calypso-stats-mobile-cta-jetpack-apple-badge';
+const URL_JETPACK_GOOGLE = 'calypso-stats-mobile-cta-jetpack-google-badge';
+const URL_JETPACK_QRCODE = 'calypso-stats-mobile-cta-jetpack-qrcode';
+const URL_WOO_A8C = 'calypso-stats-mobile-cta-woo-link';
+const URL_WOO_APPLE = 'calypso-stats-mobile-cta-woo-apple-badge';
+const URL_WOO_GOOGLE = 'calypso-stats-mobile-cta-woo-google-badge';
+const URL_WOO_QRCODE = 'calypso-stats-mobile-cta-woo-qrcode';
```

Comment:
If we defined these as a [standard Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), we wouldn't need to create `slugs` or `events` arrays below. We could use [`Map.prototype.has`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) to check membership/validity.

Reply:
I spent some time tinkering with a Map (in response to this feedback) but I couldn't quite get it working the way I wanted. While it eliminated the need for the array of CONST values, it means I then had to use raw strings at the call sites. I tried something like this:

```
// set up a Map of known slugs
const kvArray = [
  [ 'URL_JETPACK_A8C', 'calypso-stats-mobile-cta-jetpack-link' ],
  ['URL_WOO_A8C', 'calypso-stats-mobile-cta-woo-link' ]
];
const slugMap = new Map( kvArray );
// check for valid slug
function getURLForSlug( slug ) {
  if ( slugMap.has( slug ) ) {
    console.log( 'valid slug' );
  }
  return '';
}
// at call site. need to use raw strings?
const url = getURLForSlug( 'URL_JETPACK_A8C' );
console.log( url );
```

I ended up going with a configuration Object instead. It let me test for validity and avoid raw slugs at the call sites. I'll ping you from the updated PR so you can have a look. Also, please let me know if I overlooked something in my Map approach. Thanks @jsnmoon!

Reply:
Addressed in #70926.

Reply:
Happy to help! And you pretty much nailed it; this was what I was thinking (using your code snippet above as the basis):

```js
const slugMap = new Map( [
	[ 'jetpack-a8c', 'calypso-stats-mobile-cta-jetpack-link' ],
	[ 'woo-a8c', 'calypso-stats-mobile-cta-woo-link' ],
] );

function getURLForSlug( slug ) {
	if ( slugMap.has( slug ) ) {
		console.log( 'valid slug' );
	}
	return '';
}

const url = getURLForSlug( 'jetpack-a8c' );
console.log( url );
```

Note that `map.get( 'jetpackA8C' )` is not much different from `object.jetpackA8C`, which is the same as `object[ 'jetpackA8C' ]`.